client/extensions/woocommerce/app/promotions/promotion-create.js

```diff
@@ -106,9 +107,22 @@ class PromotionCreate extends React.Component {
 	}
 
 	onSave = () => {
-		const { site, promotion, translate } = this.props;
+		const { site, promotion, currency, translate } = this.props;
+		const validatingPromotion = promotion || { type: 'fixed_product' };
+		const errors = validateAll( validatingPromotion, currency, true );
+
+		if ( errors ) {
+			this.setState( () => ( { busy: false, saveAttempted: true } ) );
+			this.props.errorNotice(
+				translate( 'Invalid field data. Please correct the highlighted fields and try again.' ),
```

Comment:
Maybe without the `Invalid field data.` part? The second sentence is clean and points user directly to what is needed.